.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,63 @@
+name: Build and test
+on:
+  push:
+    paths:
+      - .github/workflows/**
```

Comment:
Nice - I do find the whitelisting approach a safer/more reliable one, even if it usually requires specifying a number more options!